.github/workflows/verify.yaml

```diff
@@ -0,0 +1,62 @@
+name: Verify
+
+on:
+  pull_request:
+    branches: [ pr/optimize-workflows ]
+
+jobs:
+  Verify:
+    name: Verify
+    runs-on: ubuntu-latest
+    steps:
+    - name: Free some disk space on runner
+      run: |
+        echo "free space before cleanup:"
+        df -h
+        sudo rm -rf /usr/share/dotnet /opt/ghc /usr/local/share/boost /usr/lib/jvm /usr/lib/firefox /opt/microsoft/powershell /opt/hostedtoolcache
+        echo "free space after cleanup:"
+        df -h
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+      id: go
+
+    - name: Use Node 16.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
+    - name: go cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Get npm cache directory
+      id: npm-cache-dir
+      run: |
+        echo "::set-output name=dir::$(npm config get cache)"
+    - name: npm cache
+      uses: actions/cache@v3
+      id: npm-cache # use this to check for `cache-hit` ==> if: steps.npm-cache.outputs.cache-hit != 'true'
+      with:
+        path: ${{ steps.npm-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
```

Comment:
think this part can be removed since we're not doing any node related ops in make verify

Reply:
updated in 
2f174eb